src/scenechange/mod.rs

```diff
@@ -104,237 +136,288 @@ impl SceneChangeDetector {
-            self.cpu_feature_level,
-          );
-          intra_costs.iter().map(|&cost| cost as u64).sum::<u64>() as f64
-            / intra_costs.len() as f64
-        },
-        move || {
-          let inter_costs = estimate_inter_costs(
-            frame2_ref2,
-            frame1,
-            self.bit_depth,
-            self.encoder_config,
-            self.sequence.clone(),
-          );
-          inter_costs.iter().map(|&cost| cost as u64).sum::<u64>() as f64
-            / inter_costs.len() as f64
-        },
-      );
+      THRESH_MIN
+        + (THRESH_MAX - THRESH_MIN)
+          * (distance_from_keyframe - min_keyint) as f64
+          / (max_keyint - min_keyint) as f64
+    };
+    let threshold = intra_cost * (1.0 - bias);
 
-      // Sliding scale, more likely to choose a keyframe
-      // as we get farther from the last keyframe.
-      // Based on x264 scenecut code.
-      //
-      // `THRESH_MAX` determines how likely we are
-      // to choose a keyframe, between 0.0-1.0.
-      // Higher values mean we are more likely to choose a keyframe.
-      // `0.4` was chosen based on trials of the `scenecut-720p` set in AWCY,
-      // as it appeared to provide the best average compression.
-      // This also matches the default scenecut threshold in x264.
-      const THRESH_MAX: f64 = 0.4;
-      const THRESH_MIN: f64 = THRESH_MAX * 0.25;
-      let distance_from_keyframe = frameno - previous_keyframe;
-      let min_keyint = self.encoder_config.min_key_frame_interval;
-      let max_keyint = self.encoder_config.max_key_frame_interval;
-      let bias = if distance_from_keyframe <= min_keyint / 4 {
-        THRESH_MIN / 4.0
-      } else if distance_from_keyframe <= min_keyint {
-        THRESH_MIN * distance_from_keyframe as f64 / min_keyint as f64
-      } else {
-        THRESH_MIN
-          + (THRESH_MAX - THRESH_MIN)
-            * (distance_from_keyframe - min_keyint) as f64
-            / (max_keyint - min_keyint) as f64
-      };
-      let threshold = intra_cost * (1.0 - bias);
-
-      ScenecutResult {
-        intra_cost,
-        threshold,
-        inter_cost,
-        has_scenecut: inter_cost > threshold,
-      }
-    }
+    ScenecutResult { intra_cost, inter_cost, threshold }
   }
 
-  fn delta_in_planes<T: Pixel>(
-    &self, plane1: &Plane<T>, plane2: &Plane<T>,
-  ) -> u64 {
+  /// Calculates delta beetween 2 planes
+  /// returns average for pixel
+  #[hawktracer(delta_in_planes)]
+  fn delta_in_planes(&self, plane1: &Plane<T>, plane2: &Plane<T>) -> f64 {
     let mut delta = 0;
+
     let lines = plane1.rows_iter().zip(plane2.rows_iter());
 
     for (l1, l2) in lines {
       let delta_line = l1
         .iter()
         .zip(l2.iter())
         .map(|(&p1, &p2)| {
-          (i16::cast_from(p1) - i16::cast_from(p2)).abs() as u64
+          (i16::cast_from(p1) - i16::cast_from(p2)).abs() as u32
         })
-        .sum::<u64>();
+        .sum::<u32>();
       delta += delta_line;
```

Comment:
We need to up-cast to u64 here to accumulate the line sums. Otherwise, we will be limited to `2^9` pixels altogether, not just in each row.
```suggestion
      delta += delta_line as u64;
```

Reply:
Fixed